### PR TITLE
dumpling: support dumping sequences

### PR DESCRIPTION
**Pull request by docsir**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/32538

Problem Summary: Support dumping MariaDB and TiDB sequences.

### What is changed and how it works?
not finished, will fix ut and it later.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by lichunzhu** [comment]
Is this ready for review?

**Comment by docsir** [comment]
>

File: dumpling/export/ir_impl.go [hunk 1]
```diff
@@ -249,16 +249,17 @@ func (td *tableData) RawRows() *sql.Rows {
...
+	table              string
+	colTypes           []*sql.ColumnType
+	selectedField      string
+	selectedLen        int
+	specCmts           []string
+	showCreateTable    string
+	showCreateView     string
+	avgRowLength       uint64
+	hasImplicitRowID   bool
+	showCreateSequence string
```
Comment (lance6716) [inline dumpling/export/ir_impl.go:262]: it can be moved under `showCreateView`

File: dumpling/export/sql.go [hunk 2]
```diff
@@ -174,6 +174,42 @@ func ShowCreateView(tctx *tcontext.Context, db *BaseConn, database, view string)
 	return createTableSQL.String(), createViewSQL.String(), nil
 }
 
+// ShowCreateTable constructs the create table SQL for a specified table
```
Comment (lance6716) [inline dumpling/export/sql.go:177]: need update comment to ShowCreateSequence

File: dumpling/export/sql.go [hunk 3]
```diff
@@ -174,6 +174,42 @@ func ShowCreateView(tctx *tcontext.Context, db *BaseConn, database, view string)
...
+	err := db.QuerySQL(tctx, handleOneRow, func() {
+		oneRow[0], oneRow[1] = "", ""
+	}, query)
+	if err != nil {
+		return "", err
+	}
+	createSequenceSQL.WriteString(oneRow[1])
+	createSequenceSQL.WriteString(";\n")
+
+	query = fmt.Sprintf("SHOW TABLE `%s`.`%s` NEXT_ROW_ID", escapeString(database), escapeString(sequence))
```
Comment (lance6716) [inline dumpling/export/sql.go:196]: could you write a comment about why we can't use NEXTVAL()?

File: dumpling/export/task.go [hunk 4]
```diff
@@ -34,6 +34,14 @@ type TaskViewMeta struct {
 	CreateViewSQL  string
 }
 
+// TaskSequenceMeta is a dumping sequence metadata task
+type TaskSequenceMeta struct {
```
Comment (lance6716) [inline dumpling/export/task.go:38]: where is this structure created?

File: dumpling/export/sql.go [hunk 5]
```diff
@@ -174,6 +174,42 @@ func ShowCreateView(tctx *tcontext.Context, db *BaseConn, database, view string)
...
+	results, err := db.QuerySQLWithColumns(tctx, []string{"NEXT_GLOBAL_ROW_ID", "ID_TYPE"}, query)
```
Comment (lichunzhu) [inline dumpling/export/sql.go:197]: We should use this sql only for TiDB here and we may need some other dump SQLs for MariaDB. But to me we can do this in next PR.

**Review by lichunzhu — COMMENTED** [review]
Could you please add an integration test for TiDB?

File: dumpling/tests/sequences/run.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+export DUMPLING_TEST_PORT=4000
+
+run_sql "drop database if exists sequences;"
+run_sql "create database sequences DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;"
+export DUMPLING_TEST_DATABASE="sequences"
+
+run_sql "create sequence s start with 1 minvalue 1 maxvalue 9223372036854775806 increment by 1 cache 1000 nocycle ENGINE=InnoDB;"
+run_sql "select nextval(s);"
+
+run_dumpling --no-sequences=false
```
Comment (lichunzhu) [inline dumpling/tests/sequences/run.sh:16]: Add integration tests for `--no-sequences` and `--no-schemas` like views tests?

**Review by lance6716 — APPROVED** [review]

File: dumpling/export/sql.go (hunk 5 above)
Comment (lichunzhu) [reply dumpling/export/sql.go:198]: For MariaDB we should use `select next_not_cached_value from `%s`.`%s`;`

File: dumpling/export/sql.go [hunk 7]
```diff
@@ -174,6 +174,43 @@ func ShowCreateView(tctx *tcontext.Context, db *BaseConn, database, view string)
 	return createTableSQL.String(), createViewSQL.String(), nil
 }
 
+// ShowCreateSequence constructs the create sequence SQL for a specified sequence
+// returns (createSequenceSQL, error)
+func ShowCreateSequence(tctx *tcontext.Context, db *BaseConn, database, sequence string) (string, error) {
```
Comment (lichunzhu) [inline dumpling/export/sql.go:179]: You can put `conf.ServerInfo` into this function

File: dumpling/export/sql.go [hunk 8]
```diff
@@ -194,21 +196,35 @@ func ShowCreateSequence(tctx *tcontext.Context, db *BaseConn, database, sequence
...
+		}
+		query := fmt.Sprintf("SELECT NEXT_NOT_CACHED_VALUE FROM `%s`.`%s`", escapeString(database), escapeString(sequence))
+		err := db.QuerySQL(tctx, handleOneRow1, func() {
+			oneRow1 = ""
+		}, query)
+		if err != nil {
+			return "", err
+		}
+		nextNotCachedValue, _ = strconv.ParseInt(oneRow1, 10, 64)
+		fmt.Fprintf(&createSequenceSQL, "DO SETVAL(`%s`,%d);\n", escapeString(sequence), nextNotCachedValue)
```
Comment (lichunzhu) [inline dumpling/export/sql.go:227]: Nit: Maybe use `SELECT SETVAL` as TiDB?

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM, great job!!

**Review by lichunzhu — APPROVED** [review]

**Pull request merged**